ext/ScreenShot/ScreenShot.h

```diff
@@ -5,27 +5,151 @@
+
+			irr::video::IFrameBuffer* createDefaultFBOForScreenshoting(core::smart_refctd_ptr<IrrlichtDevice> device)
+			{
+				auto driver = device->getVideoDriver();
+
+				auto createAttachement = [&](bool colorBuffer)
+				{
+					asset::ICPUImage::SCreationParams imgInfo;
+					imgInfo.format = colorBuffer ? asset::EF_R8G8B8A8_SRGB : asset::EF_D24_UNORM_S8_UINT;
+					imgInfo.type = asset::ICPUImage::ET_2D;
+					imgInfo.extent.width = driver->getCurrentRenderTargetSize().Width;
+					imgInfo.extent.height = driver->getCurrentRenderTargetSize().Height;
+					imgInfo.extent.depth = 1u;
+					imgInfo.mipLevels = 1u;
+					imgInfo.arrayLayers = 1u;
+					imgInfo.samples = asset::ICPUImage::ESCF_1_BIT;
+					imgInfo.flags = static_cast<asset::IImage::E_CREATE_FLAGS>(0u);
+
+					auto image = asset::ICPUImage::create(std::move(imgInfo));
+					const auto texelFormatBytesize = getTexelOrBlockBytesize(image->getCreationParameters().format);
+
+					auto texelBuffer = core::make_smart_refctd_ptr<asset::ICPUBuffer>(image->getImageDataSizeInBytes());
+					auto regions = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<asset::ICPUImage::SBufferCopy>>(1u);
+					asset::ICPUImage::SBufferCopy& region = regions->front();
+
+					region.imageSubresource.mipLevel = 0u;
+					region.imageSubresource.baseArrayLayer = 0u;
+					region.imageSubresource.layerCount = 1u;
+					region.bufferOffset = 0u;
+					region.bufferRowLength = image->getCreationParameters().extent.width;
+					region.bufferImageHeight = 0u;
+					region.imageOffset = { 0u, 0u, 0u };
+					region.imageExtent = image->getCreationParameters().extent;
+
+					image->setBufferAndRegions(std::move(texelBuffer), regions);
+
+					asset::ICPUImageView::SCreationParams imgViewInfo;
+					imgViewInfo.image = std::move(image);
+					imgViewInfo.format = colorBuffer ? asset::EF_R8G8B8A8_SRGB : asset::EF_D24_UNORM_S8_UINT;
+					imgViewInfo.viewType = asset::IImageView<asset::ICPUImage>::ET_2D;
+					imgViewInfo.flags = static_cast<asset::ICPUImageView::E_CREATE_FLAGS>(0u);
+					imgViewInfo.subresourceRange.baseArrayLayer = 0u;
+					imgViewInfo.subresourceRange.baseMipLevel = 0u;
+					imgViewInfo.subresourceRange.layerCount = imgInfo.arrayLayers;
+					imgViewInfo.subresourceRange.levelCount = imgInfo.mipLevels;
+
+					auto imageView = asset::ICPUImageView::create(std::move(imgViewInfo));
+					auto gpuImageView = driver->getGPUObjectsFromAssets(&imageView.get(), &imageView.get() + 1)->front();
+
+					return std::move(gpuImageView);
+				};
+
+				auto gpuImageViewDepthBuffer = createAttachement(false);
+				auto gpuImageViewColorBuffer = createAttachement(true);
+
+				auto frameBuffer = driver->addFrameBuffer();
+				frameBuffer->attach(video::EFAP_DEPTH_ATTACHMENT, std::move(gpuImageViewDepthBuffer));
+				frameBuffer->attach(video::EFAP_COLOR_ATTACHMENT0, std::move(gpuImageViewColorBuffer));
+
+				return frameBuffer;
+			};
+
+			/*
+				Create a ScreenShot with gpu image usage.
```

Comment:
because of the fence waiting required to do this all in one function, you should warn in the documentation that the function may and will stall the CPU to wait on the GPU to finish